test/e2e/azure_logcollector.go

```diff
@@ -108,14 +109,18 @@ func (k AzureLogCollector) CollectMachinePoolLog(ctx context.Context, management
 	}
 
 	for i, instance := range mp.Spec.ProviderIDList {
-		hostname := mp.Status.NodeRefs[i].Name
+		if mp.Status.NodeRefs != nil && len(mp.Status.NodeRefs) >= (i+1) {
+			hostname := mp.Status.NodeRefs[i].Name
 
-		if err := collectLogsFromNode(ctx, managementClusterClient, cluster, hostname, isWindows, filepath.Join(outputPath, hostname)); err != nil {
-			errors = append(errors, err)
-		}
+			if err := collectLogsFromNode(ctx, managementClusterClient, cluster, hostname, isWindows, filepath.Join(outputPath, hostname)); err != nil {
+				errors = append(errors, err)
+			}
 
-		if err := collectVMSSBootLog(ctx, instance, filepath.Join(outputPath, hostname)); err != nil {
-			errors = append(errors, err)
+			if err := collectVMSSBootLog(ctx, instance, filepath.Join(outputPath, hostname)); err != nil {
+				errors = append(errors, err)
+			}
+		} else {
+			Logf(fmt.Sprintf("INFO: MachinePool instance %s does not have a corresponding NodeRef", instance))
```

Comment:
can we add something like "skipping node and boot log collection for MachinePool instance %: " to the log message so the impact is more obvious in the output?

Reply:
done

Reply:
sorry if it wasn't clear, I meant in addition to the message you already had that way it's clear what we're doing and why

Reply:
all good, done